### DISCUSSION TITLE
Prioritizing the caption field

### DISCUSSION THREAD

**Discussion by alexjerg** [archetyped/simple-lightbox#1061] [Q&A]
Hey All,
I love this plugin, one minor thing I'm running into is we use the media library to add the image title, alt text, caption, and description.

Is it possible to have SLB prioritize the Caption field from WordPress' media library first? I know that the title precedence is set to prioritize that 3rd, but I was hoping to have it use the caption always as all of our images that will use the SLB will have captions included. Right now, if I don't have the alternative text manually set on the page in which the image is added, no caption will display at all—despite having one set in the media library.

**Comment by archetyped** [comment]
Actually, by default the Media Library's "Caption" field *is* what would be used by SLB in the lightbox.  The values with a higher priority require manual modification (e.g. directly editing the link's HTML to include a `title` attribute).

If you are experiencing different behavior, please provide:

1. A link to a page that exhibits the behavior you are seeing.
2. A screenshot of the media library entry for an image on the page provided.

> Right now, if I don't have the alternative text manually set on the page in which the image is added, no caption will display at all—despite having one set in the media library.

Note that as per [SLB's Metadata documentation](https://github.com/archetyped/simple-lightbox/wiki/Metadata#title-precedence), SLB references the "image's `alt` attribute", not the value from the Media Library's "Alt text" field.

When inserting an image block, WordPress automatically sets the value of its "Alternative text" field from the Media Library's "Alt text" field.  However, once the image block is created, these these fields are independent of each other.  As such, changing the "Alt text" field in the Media Library has no effect on the `alt` attribute of previously-inserted images.  The block's own "Alternative text" needs to set directly.
